openedx/features/edly/middleware.py

```diff
@@ -27,6 +33,26 @@ def process_request(self, request):
         if request.user.is_superuser or request.user.is_staff:
             return
 
+        try:
+            edly_sub_org = EdlySubOrganization.objects.get(
+                Q(lms_site=request.site) |
+                Q(studio_site=request.site) |
+                Q(preview_site=request.site)
+            )
```

Comment:
we can avoid querying the db on each request by writing is this way:

```edly_sub_org = None
edly_sub_org = getattr(self.request.site, 'edly_sub_org_for_studio', edly_sub_org)
edly_sub_org = getattr(self.request.site, 'edly_sub_org_for_lms', edly_sub_org)
edly_sub_org = getattr(self.request.site, 'edly_sub_org_for_preview', edly_sub_org)
```

Reply:
Done